### ISSUE TITLE
Slow shutdown, when a pod is terminated it takes around ~30 secs for the PHPA to stop

### ISSUE THREAD

**Issue by jthomperoo** [jthomperoo/predictive-horizontal-pod-autoscaler#40]
**Describe the bug**
Slow shutdown, when a pod is terminated it takes around ~30 secs for the PHPA to stop. I think this is because of the entrypoint shell script that runs the PHPA in the docker image - the `SIGTERM` signal is not propagated onto the Custom Pod Autoscaler binary, it does not know it is shutting down so it keeps running. The 30 second wait is the grace period before K8s sends a `SIGKILL` which forces the PHPA to stop.

**To Reproduce**
1. Deploy a PHPA, for example `simple-linear` in the `examples/` directory.
2. Wait until it is running with `kubectl get pods`.
3. Watch the PHPA logs, for example `kubectl logs simple-linear-example --follow`.
3. Delete the PHPA using , for example `kubectl delete cpa simple-linear-example`.
4. Watch the pods and PHPA logs, it takes around 30 seconds to terminate, and the logs do not print anything about shutting down.

**Expected behavior**
PHPA should shut down promptly, in line with a normal CPA, and should print out `Shutting down...` to the logs when the `SIGTERM` signal is received. As is expected and defined in the CPA here:
https://github.com/jthomperoo/custom-pod-autoscaler/blob/d00bb5f72d382ba07f9b23437e2f8ce844edd18c/cmd/custom-pod-autoscaler/main.go#L239-L247

Basically instead of the `build/entrypoint.sh` script starting the binary using:

```bash
/cpa/custom-pod-autoscaler
```

It should use:

```bash
exec /cpa/custom-pod-autoscaler
```

(As explained here: https://hynek.me/articles/docker-signals/)
